File: Languages/C/Day022.c
// Languages - C - Day 22
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that removes all duplicate characters from a string while preserving the original order.


#include <stdio.h>
#include <string.h>
#include <stdbool.h>

// removeDuplicates removes duplicate characters from a string while preserving the original order.
void removeDuplicates(char* input, char* output) {
    bool seen[256] = { false };
    int j = 0;
    for (int i = 0; input[i] != '\0'; i++) {
        if (!seen[(unsigned char)input[i]]) {
            seen[(unsigned char)input[i]] = true;
            output[j++] = input[i];
        }
    }
    output[j] = '\0';
}

int main() {
    char input[100];
    char output[100];
    printf("Enter a string: ");
    fgets(input, sizeof(input), stdin);
    input[strcspn(input, "\n")] = '\0'; // Remove newline character
    if (strlen(input) == 0) {
        printf("Input cannot be empty.\n");
        return 1;
    }
    removeDuplicates(input, output);
    printf("String after removing duplicates: %s\n", output);
    return 0;
}

File: Languages/PHP/Day022.php
<!-- Languages - PHP - Day 22 -->
<!-- Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables. -->

<!-- Create a function that removes all duplicate characters from a string while preserving the original order.-->


<?php

// Function to remove duplicate characters while preserving order
function removeDuplicates($input) {
    if (empty(trim($input))) {
        return "Input cannot be empty.";
    }
    $seen = [];
    $result = '';
    for ($i = 0; $i < strlen($input); $i++) {
        $char = $input[$i];
        if (!isset($seen[$char])) {
            $seen[$char] = true;
            $result .= $char;
        }
    }
    return $result;
}

// Get user input and display the result
echo "Enter a string:\n";
$input = trim(fgets(STDIN));

if (empty($input)) {
    echo "Input cannot be empty.\n";
    exit;
}

$result = removeDuplicates($input);
echo "String after removing duplicates: $result\n";

?>
